scripts/robust.yaml

```diff
@@ -0,0 +1,62 @@
+id: str(min=7, max=7)
+title: str(min=1, max=160)
+description: str(min=1)
+classification: str(min=1)
+keywords: list(str(min=1, max=80))
+system: str(min=1, max=40)
+severity: enum("error", "warning", "convention-violation", "bad-smell",
+            "minor-issue", "not-a-bug")
+links: list(str())
+bug: include('_bug')
+fix: include('_fix')
+time-machine: include('_time-machine')
+bugzoo: include('_bugzoo')
+
+---
+
+_bug:
+    phase: enum("programming-time", "build-time", "compile-time",
+            "deployment-time", "runtime-initialization", "runtime-operation")
+    specificity: enum("general issue", "robotics-specific", "ROS-specific",
+            "application-specific")
+    architectural-location: enum("application-specific code", "platform code", "N/A")
+    application: any(str(min=1), null())
+    task: enum("perception", "localization", "planning", "manipulation",
+            "human-robot interaction", "simulation", "diagnostics", "SLAM",
+            "N/A")
+    subsystem: str(min=1)
```

Comment:
Shouldn't this be an enum?

Reply:
you mean the subsystem.  I think we should definitely make sure that consistent names and spelling is used, otherwise, I think it is of secondary importance, whether it is an enum or not. If you make it enum it is easier for you to detect divergences.  The cost is that the scheme has to be changed whenever a new value pops up. I think this is a price that is entirely acceptable.  So if you prefer to let yamale control it, just go ahead.